### PR TITLE
[react] Add types for `inert` to Experimental release channel

### PR DESCRIPTION
**Pull request by eps1lon**
Waiting for release of https://github.com/facebook/react/pull/24730

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react/pull/24730
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react@latest/))
was missing the following properties:
1. unstable_act
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9a238d12d8f275ff38c5608ce1c782b36eebbdee
</p>

File: types/react/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for React 18.0
+// Type definitions for React 18.3
```
Comment (votemike) [inline types/react/index.d.ts:1]: Is `inert` only available in React 18.3?

File: types/react/index.d.ts [hunk 2]
```diff
@@ -1856,6 +1856,10 @@ declare namespace React {
         draggable?: Booleanish | undefined;
         hidden?: boolean | undefined;
         id?: string | undefined;
+        /**
+         * @see https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert
```
Comment (votemike) [inline types/react/index.d.ts:1860]: Do we need a comment in here? None of the other attributes have comments.

File: types/react/index.d.ts (hunk 1 above)
Comment (eps1lon) [reply types/react/index.d.ts:1]: Please understand that this PR is in draft and waiting for a resolution of https://github.com/facebook/react/pull/24730

File: types/react/index.d.ts (hunk 2 above)
Comment (eps1lon) [reply types/react/index.d.ts:1860]: > Do we need a comment in here?

It's JSDOC which means it appears in editors supporting it. This helps people understand what this attribute is for.


> None of the other attributes have comments.

I don't know why for each. For some there's no useful MDN article. For some people might not have been aware how useful JSDOC can be

**Comment by JeroenReumkens** [comment]
Would love to see this added now support for `inert` and `dialog` is so good. `inert` is very useful for dialogs, and having proper types hopefully makes more people use this!

**Pull request merged**